cns/restserver/restserver.go

```diff
@@ -1607,3 +1595,433 @@ func (service *HTTPRestService) getNumberOfCPUCores(w http.ResponseWriter, r *ht
+		req           cns.DeleteCompartmentWithNCsRequest
+	)
+
+	err = service.Listener.Decode(w, r, &req)
+	log.Request(service.Name, &req, err)
+
+	if err != nil {
+		returnMessage = fmt.Sprintf("[Azure CNS] ERROR: Unable to decode input request")
+		returnCode = InvalidParameter
+	} else {
+		switch r.Method {
+		case "DELETE":
+			if req.CompartmentID <= hnsclient.DefaultNetworkCompartmentID {
+				returnMessage = fmt.Sprintf("ERROR: Invalid compartment ID specified")
+				returnCode = InvalidParameter
+			} else {
+				if compartmentNCData, err := service.getCompartmentNCData(req.CompartmentID); err == nil {
+					for _, endpointName := range compartmentNCData {
+						if err = hnsclient.CleanupEndpoint(endpointName); err != nil {
+							returnMessage = err.Error()
+							returnCode = UnexpectedError
+							break
+						}
+					}
+
+					if err == nil {
+						if err = hnsclient.DeleteCompartment(req.CompartmentID); err != nil {
+							returnMessage = err.Error()
+							returnCode = UnexpectedError
+						} else {
+							// Compartment deleted successfully
+							returnMessage = fmt.Sprintf("Successfully deleted network compartment "+
+								"with ID: %d", req.CompartmentID)
+							service.deleteCompartmentNCData(req.CompartmentID)
+						}
+					}
+				} else {
+					returnMessage = fmt.Sprintf("ERROR: Attempting to delete invalid compartment")
+					returnCode = InvalidParameter
+				}
+			}
+		default:
+			returnMessage = fmt.Sprintf("ERROR: deleteCompartmentWithNCs API expects a DELETE")
+			returnCode = UnsupportedVerb
+		}
+	}
+
+	resp := &cns.Response{
+		ReturnCode: returnCode,
+		Message:    returnMessage,
+	}
+
+	err = service.Listener.Encode(w, &resp)
+	log.Response(service.Name, resp, resp.ReturnCode, ReturnCodeToString(resp.ReturnCode), err)
+}
+
+// Handles requests to create compartment with NCs.
+func (service *HTTPRestService) createCompartmentWithNCs(w http.ResponseWriter, r *http.Request) {
+	log.Printf("[Azure CNS] createCompartmentWithNCs")
+
+	var (
+		returnCode    int
+		returnMessage string
+		err           error
+		compartmentID int
+		req           cns.CreateCompartmentWithNCsRequest
+	)
+
+	err = service.Listener.Decode(w, r, &req)
+	log.Request(service.Name, &req, err)
+
+	if err != nil {
+		returnMessage = fmt.Sprintf("[Azure CNS] ERROR: Unable to decode input request")
+		returnCode = InvalidParameter
+	} else {
+		switch r.Method {
+		case "POST":
+			if len(req.NCIDs) <= 0 {
+				returnMessage = fmt.Sprintf("ERROR: No NCID specified in the request")
+				returnCode = InvalidParameter
+			} else if len(req.NCIDs) > hnsclient.MaxNCsPerCompartment {
+				returnMessage = fmt.Sprintf("ERROR: Number of NCIDs specified (%d) exceeds max supported (%d)",
+					len(req.NCIDs), hnsclient.MaxNCsPerCompartment)
+				returnCode = InvalidParameter
+			} else {
+				if compartmentID, err = hnsclient.CreateCompartment(); err != nil {
+					returnMessage = err.Error()
+					returnCode = UnexpectedError
```

Comment:
is this needed?

Reply:
yes, otherwise we won't set the correct error message and return code as by default we set those to empty